### ISSUE TITLE
Issue previewing unpublished content

### ISSUE THREAD

**Issue by costabl** [acquia/next-acms#11]
The issue is with previewing unpublished content, the preview within Drupal returns a 404.

1. Unpublish any content, I'm using Place but the issue is the same with any content type.

2. Preview within Drupal using the iframe

3. Copy the iframe URL and  open directly in the browser

4. It works fine when opening directly in the browser and it shows the preview mode message.

**Comment by harumijang** [comment]
Lines 123-127 in `[[...slug]].tsx` causes this

```
  // If we're not in preview mode and the resource is not published,
  // Return page not found.
  if (!context.preview && node?.status === false) {
    return {
      notFound: true,
    };
  }
```

**Comment by harumijang** [comment]
Found that `context.preview` is undefined when in development mode but has the correct value when in production mode and therefore shows the preview.

**Comment by harumijang** [comment]
I think this might be the reason why? Or at the very least it was a helpful pointer for me. 

https://github.com/vercel/next.js/discussions/11315#discussioncomment-285086

**Comment by lauriii** [comment]
Based on the research done by @harumijang, I'm wondering if this is something that should be addressed by documenting that the preview mode doesn't work with the dev mode.

**Comment by lauriii** [comment]
I filed PR which makes preview mode work on development. It should address the confusing behavior where future revisions were not taken into account when Next.js was rendered on the iframe.

I also noticed that rendering future revisions in the preview would require some additional permissions added in ACMS. We should file PR against ACMS that adds `view any unpublished content` and `view latest version` permissions for the headless user.
